frontend/services/src/imageService.ts

```diff
@@ -1,89 +1,110 @@
-        console.error("Error fetching images:", error);
-        return [];
-    }
+  const eventRef = ref(storage, "events/" + eventID);
+  let itemArray: string[] = [];
+
+  try {
+    const res = await listAll(eventRef);
+    res.items.forEach((itemRef) => {
+      itemArray.push(itemRef.fullPath);
+    });
+    return itemArray;
+  } catch (error) {
+    imageServiceLogger.error(`Error fetching images ${error}`);
+    return [];
+  }
 }
 
 export async function getEventImageUrls(eventID: string): Promise<string[]> {
-    const imageRefs = await getEventImageLocation(eventID);
-    try {
-        const urlPromises = imageRefs.map((imageRef) => {
-            const refPath = ref(storage, imageRef);
-            return getDownloadURL(refPath);
-        });
-
-        const urls = await Promise.all(urlPromises);
-
-        return urls;
-    } catch (error) {
-        console.error("Error fetching download URLs:", error);
-        return [];
-    }
+  const imageRefs = await getEventImageLocation(eventID);
+  try {
+    const urlPromises = imageRefs.map((imageRef) => {
+      const refPath = ref(storage, imageRef);
+      return getDownloadURL(refPath);
+    });
+
+    const urls = await Promise.all(urlPromises);
+
+    return urls;
+  } catch (error) {
+    imageServiceLogger.error(`Error fetching download URLs ${error}`);
+    return [];
+  }
+}
+
+export async function uploadProfilePhoto(userID: string, file: File): Promise<string> {
+  const imagePath = `users/${userID}/profilepicture${generateImageId()}`;
+  const imageRef = ref(storage, imagePath);
+  const url = await uploadImage(imageRef, file);
+
+  return url;
 }
 
 export async function uploadUserImage(userID: string, file: File): Promise<string> {
-    const timestamp = Date.now(); // To ensure unique filenames
-    const imagePath = `users/${userID}/${timestamp}_${file.name}`;
-    const imageRef = ref(storage, imagePath);
-
-    try {
-        // Upload the file to Firebase Storage
-        await uploadBytes(imageRef, file);
-
-        // Get the download URL of the uploaded file
-        const url = await getDownloadURL(imageRef);
-
-        return url;
-    } catch (error) {
-        console.error("Error uploading image:", error);
-        throw error; // Re-throw the error so the caller can handle it
-    }
+  const imagePath = `users/${userID}/${generateImageId()}`;
+  const imageRef = ref(storage, imagePath);
+  const url = await uploadImage(imageRef, file);
+
+  return url;
 }
 
+async function uploadImage(imageRef: StorageReference, file: File): Promise<string> {
+  try {
+    // Upload the file to Firebase Storage
+    await uploadBytes(imageRef, file);
 
+    // Get the download URL of the uploaded file
+    const url = await getDownloadURL(imageRef);
+
+    return url;
+  } catch (error) {
+    imageServiceLogger.error(`Error uploading image ${error}`);
+    throw error; // Re-throw the error so the caller can handle it
+  }
+}
+
+function generateImageId(): string {
```

Comment:
amazing thanks for functioning everything, its much neater and much more manageable!